platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/AbstractArtifactRepository.java

```diff
@@ -56,7 +59,7 @@ public abstract class AbstractArtifactRepository implements ArtifactRepositoryIn
     private Action<? super ActionConfiguration> componentMetadataSupplierRuleConfiguration;
     private Action<? super ActionConfiguration> componentMetadataListerRuleConfiguration;
     private final ObjectFactory objectFactory;
-    private final Supplier<RepositoryContentDescriptorInternal> repositoryContentDescriptor = Suppliers.memoize(this::createRepositoryDescriptor)::get;
+    private final Supplier<RepositoryContentDescriptorInternal> repositoryContentDescriptor = Lazy.locking().of(this::createRepositoryDescriptor);
```

Comment:
❓ What's so expensive about creating this descriptor that it needs to be memo-ized?  These object to be created here all seem pretty lightweight, and the `PluginArtifactRepository` doesn't memo-ize.  Should this be changed to non-lazy?

Reply:
The memoization was added here: https://github.com/gradle/gradle/commit/4a0be9c16aaae2fc779946e2e0fe5515499a417b

The commit message and associated PR do not hint towards why this change was made. 

In practice, it seems this type is always realized. I'll un-memoize and see if it affects performance tests.